greenswitch/esl.py

```diff
@@ -458,6 +459,13 @@ def uuid_break(self):
             raise OutboundSessionHasGoneAway
         self.send('api uuid_break %s' % self.uuid)
 
+    def raise_if_disconnected(self):
+        """This function will raise esl.OutboundSessionHasGoneAway exception
+        if the session is not connected
+        """
+        if not self._outbound_connected:
```

Comment:
@italorossi what do you think about use `self._lingering` instead of `self. _outbound_connected` and during `on_hangup` set the `self._lingering` to True?

```python
# We're not allowed to send more commands.
        # lingering True means we already received a hangup from the caller
        # and any commands sent at this time to the session will fail
        if self._lingering:
            raise OutboundSessionHasGoneAway()
```

https://github.com/EvoluxBR/greenswitch/blob/b04c8988b52ee26da1983b2c98ffed661e676f86/greenswitch/esl.py#L334